src/tensorflow.cc

```diff
@@ -1332,62 +1340,264 @@ AutoCompleteHelper::FixBatchingSupport()
   return nullptr;  // success
 }
 
+TRITONTF_IO*
+FindModelIOByName(
+    const char* name, const std::vector<const TRITONTF_IOList*> list)
+{
+  if (name == nullptr) {
+    return nullptr;
+  }
+
+  for (const auto item : list) {
+    TRITONTF_IO* io = item->io_;
+    if (!strcmp(name, io->name_)) {
+      return io;
+    }
+  }
+
+  return nullptr;
+}
+
+TRITONTF_IO*
+FindModelIOByName(const char* name, const TRITONTF_IOList* list)
+{
+  if (name == nullptr) {
+    return nullptr;
+  }
+
+  while (list) {
+    TRITONTF_IO* io = list->io_;
+    if (!strcmp(name, io->name_)) {
+      return io;
+    }
+    list = list->next_;
+  }
+
+  return nullptr;
+}
+
+
 TRITONSERVER_Error*
-AutoCompleteHelper::FixIOConfig(
-    const TRITONTF_IOList* reference_list, const char* key)
+AutoCompleteHelper::FillMissingValues(
+    TRITONTF_IO* io, triton::common::TritonJson::Value& config)
 {
-  // Replace I/O even if inputs / outputs are specified in config.
-  triton::common::TritonJson::Value ios;
-  bool found_ios = model_state_->ModelConfig().Find(key, &ios);
+  triton::common::TritonJson::Value tmp(
+      model_state_->ModelConfig(),
+      triton::common::TritonJson::ValueType::OBJECT);
+
+  bool found_name = config.Find("name", &tmp);
+  if (!found_name) {
+    triton::common::TritonJson::Value name;
+    RETURN_IF_ERROR(config.AddString("name", io->name_));
+  }
+
+  bool found_config_data_type = config.Find("data_type", &tmp);
+  std::string data_type_str;
+  tmp.AsString(&data_type_str);
+  bool should_auto_complete_data_type =
+      !found_config_data_type || DataTypeIsInvalid(data_type_str);
+  if (should_auto_complete_data_type) {
+    config.SetString("data_type", ConvertToModelConfigString(io->data_type_));
```

Comment:
SetStringObject is not in the TritonJson API. This was a new method which was in the 'common' PR for this task. IMO this was the cleaner option but I am willing to fix if there is something I missed.

Reply:
Yes. I know the PR. I was just wondering if we can rename the function as it sets/override an object within the caller object. This would mean I am requesting a renaming in common repo as well.

Reply:
I misunderstood your comment. Yes, I can update the method signature here.